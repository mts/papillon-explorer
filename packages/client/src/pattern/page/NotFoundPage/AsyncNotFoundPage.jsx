import React from 'react'

const LazyNotFoundPage = React.lazy(() => import('./NotFoundPage'))

export const AsyncNotFoundPage = () => (
  <React.Suspense fallback={null}>
    <LazyNotFoundPage />
  </React.Suspense>
)
