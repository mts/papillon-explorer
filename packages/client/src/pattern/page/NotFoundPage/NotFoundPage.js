import React from 'react'
import { Query } from 'react-apollo'
import { PAGES } from '../../../graphql/query'
import NotFoundTemplate from '../../template/NotFoundTemplate'

const NotFoundPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? <NotFoundTemplate page={data.pages.pages.find(page => page.id === 'not-found')} /> : null
      }
    </Query>
  )
}

export default NotFoundPage
