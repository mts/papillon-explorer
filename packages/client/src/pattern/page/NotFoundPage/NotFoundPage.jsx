import React from 'react'
import { object } from 'prop-types'
import { useQuery } from '@apollo/react-hooks'
import { PAGES } from '../../../graphql/query'
import { NotFoundTemplate } from '../../template/NotFoundTemplate'

export const NotFoundPage = ({ location }) => {
  const { data } = useQuery(PAGES)

  if (data?.pages?.pages) {
    return <NotFoundTemplate page={data.pages.pages.find(page => page.id === 'not-found')} location={location} />
  }

  return null
}

NotFoundPage.propTypes = {
  location: object.isRequired,
}

export default NotFoundPage
