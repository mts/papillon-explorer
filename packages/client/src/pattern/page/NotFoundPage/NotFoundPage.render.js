import React from 'react'
import NotFoundPage from './NotFoundPage'

const notFoundPageDefault = <NotFoundPage />

export default notFoundPageDefault
