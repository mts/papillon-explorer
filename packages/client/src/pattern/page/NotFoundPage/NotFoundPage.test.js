import React from 'react'
import { notFoundPageDefault } from './NotFoundPage.render'

jest.mock('@apollo/react-hooks', () => {
  return {
    useQuery: () => {
      return {
        data: {
          pages: {
            pages: [{ id: 'not-found', name: 'some-name' }],
          },
        },
      }
    },
  }
})

jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

describe('<NotFoundPage />', () => {
  describe('Snaphot', () => {
    test('must match notFoundPageDefault', () => {
      expect(global.renderToJSON(notFoundPageDefault)).toMatchSnapshot()
    })
  })
})
