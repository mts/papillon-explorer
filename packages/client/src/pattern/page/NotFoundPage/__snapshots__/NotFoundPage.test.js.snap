// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> Snaphot must match notFoundPageAsync 1`] = `null`;

exports[`<NotFoundPage /> Snaphot must match notFoundPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  <div>
    <h3>
      No match for 
      <code>
        some-path-name
      </code>
    </h3>
  </div>
  <div>
    some link
  </div>
</div>
`;

exports[`<NotFoundPage /> Snaphot must match notFoundPageNull 1`] = `null`;
